### PR TITLE
testing: mac os

### PR DESCRIPTION
**Pull request by oliver-sanders**
@kinow you said on Riot that you had hit an issue with Isodatetime and Mac OS which surprised me since isodatetime shouldn't interface with the OS directly.

I've worked on Mac OS builds with Travis before so thought I'd quickly add one here just to make sure. Feel free to reject if we don't think it's worth keeping.

### CODE REVIEW

**Comment by matthewrmshin** [comment]
Looks good. Waiting for test to complete.

**Comment by matthewrmshin** [comment]
The MacOS stage says "no language set" instead of "python ..."?

File: .travis.yml [hunk 1]
```diff
@@ -18,8 +18,21 @@ matrix:
       env: RUN_COVERAGE=false TZ=UTC
     - python: 3.8-dev
       env: RUN_COVERAGE=false
+    - os: osx
+      env: RUN_COVERAGE=false
+      # Install Python via Homebrew, Travis cannot install Python ATM.
+      language: generic
```
Comment (oliver-sanders) [inline .travis.yml]: > The MacOS stage says "no language set" instead of "python ..."?

That would be because of this line.

**tldr;** *Travis can't install different python versions automatically you need to provision it yourself.*

**long overdue rant;** As is well known Python environments usually become a bit of a mess. On Mac OS the problem becomes slightly messier as Mac OS does not actually have a package manager, well not an official one anyway. Apple gives you system Python but if you want anything else there isn't an Apple provided solution to getting it. The on-official but incredibly popular Brew package manager might as well be dubbed "the" Mac OS package manager. Most people turn to Brew to install different Python versions, consequently the Brew formula is pretty good and actually surpasses the system installation in some ways. So Brew Python is what everyone should use but neither it nor Brew are actually installed by default leaving Travis in a slightly sticky position. Their solution for the moment is to do nothing which is great except that if you try to specify a Python version it just dies in a nonsensical way. The workaround to all this mess is to set the language to something other than Python and just install Brew Python yourself.

**Comment by matthewrmshin** [comment]
The test has passed after being kicked...

Do we want to have a look at the hanging test problem before merge or after?

**Comment by kinow** [comment]
>Do we want to have a look at the hanging test problem before merge or after?

Good question. Will leave it up to @oliver-sanders. If happy with the current approach, feel free to merge, or move the discussion to an issue, etc.

**Comment by oliver-sanders** [comment]
I'm going to kick Travis a few times to see if it hangs again...

**Comment by oliver-sanders** [comment]
OK, kicked a few times, can't replicate. Merging.
